src/Core/UiFont.re

```diff
@@ -1,37 +1,17 @@
 type t = {
-  fontFile: string,
-  fontFileItalic: string,
-  fontFileBold: string,
-  fontFileSemiBold: string,
-  fontFileSemiBoldItalic: string,
-  fontSize: float,
+  normal: Revery.Font.Family.t,
+  italic: Revery.Font.Family.t,
+  bold: Revery.Font.Family.t,
+  semiBold: Revery.Font.Family.t,
+  semiBoldItalic: Revery.Font.Family.t,
+  size: float,
 };
 
-let create =
-    (
-      ~fontFile,
-      ~fontFileBold,
-      ~fontFileItalic,
-      ~fontFileSemiBold,
-      ~fontFileSemiBoldItalic,
-      ~fontSize,
-      (),
-    ) => {
-  fontFile,
-  fontFileBold,
-  fontFileItalic,
-  fontFileSemiBold,
-  fontFileSemiBoldItalic,
-  fontSize,
+let default = {
+  normal: Revery.Font.Family.fromFile("Inter-UI-Regular.ttf"),
+  bold: Revery.Font.Family.fromFile("Inter-UI-Bold.ttf"),
+  italic: Revery.Font.Family.fromFile("Inter-UI-Italic.ttf"),
+  semiBold: Revery.Font.Family.fromFile("Inter-UI-Medium.ttf"),
+  semiBoldItalic: Revery.Font.Family.fromFile("Inter-UI-MediumItalic.ttf"),
+  size: 12.,
 };
```

Comment:
Just loading all the variants separately seem to work fine. Not entirely sure how to combine these into a single family.

Reply:
The `fromFile` function takes an optional variant resolver. You can do something like:

```ocaml
let family = Font.Family.fromFile(~variant=(~weight, ~italicized, ~monospaced as _) =>
  switch ((weight, italicized)) {
  | (Weight.Bold, false) => "Inter-UI-Bold.ttf"
  ...
  | _ => "Inter-UI-Regular.ttf"
  }, "Inter-UI-Regular.ttf")
```

Reply:
I have a suspicion changing this may also fix that weird line bug you were seeing

Reply:
Oh, neat! That API is a bit confusing though, with the default case specified outside, and one of the solver cases therefore ignored. A separate `fromFiles` function would be better I think. I might try to tackle that and the few other minor API issues I've noticed before I do this.